### ISSUE TITLE
Apparent docket and opinion_cluster data mismatch

### ISSUE THREAD

**Issue by taylorkangbeck** [freelawproject/courtlistener#3218]
I'm trying to filter opinions by jurisdiction, and in order to do so I need to join search_court and search_opinion via search_docket and search_opinioncluster. But when manually inspecting the results, I find multiple rows where search_docket.case_name and search_opinioncluster.case_name significantly differ (ie refer to different cases).

An example on the Courtlistener website: https://www.courtlistener.com/docket/2372379/govan-v-director-tdcj/
In my join, `search_docket.case_name = 'Govan v. Director TDCJ'`, and `search_opinioncluster.case_name = 'Jacko v. Thorn Americas, Inc.'` These cases seem to have nothing to do with one another, with Govan being a prisoner/death row case and Jacko being litigation against a storage company.

This is not uncommon. Perhaps I'm missing something, but it seems to be a data issue.

**Comment by mlissner** [comment]
Yeah, this is definitely a data issue. It looks like in this instance we joined 4:00-cv-424 from one and 1:00-cv-424 from the other. 

We will need to clean this up, thank you. Are you finding these in some kind of automated way?

**Comment by mlissner** [comment]
@flooie let's get this on your radar for review and fix.
